lib/Sema/MiscDiagnostics.cpp

```diff
@@ -6229,6 +6529,7 @@ void swift::performSyntacticExprDiagnostics(const Expr *E,
   diagSyntacticUseRestrictions(E, DC, isExprStmt);
   diagRecursivePropertyAccess(E, DC);
   diagnoseImplicitSelfUseInClosure(E, DC);
+  diagnoseImplicitWeakToStrongCapture(E, DC);
```

Comment:
does the order here matter?

Reply:
No it shouldn't, MiscDiagnostics passes should never mutate the AST